egs/wsj/s5/utils/prepare_lang.sh

```diff
@@ -285,26 +320,25 @@ if "$silprob"; then
     }' > $tmpdir/lexiconp.txt
 fi
 
-cat $tmpdir/lexiconp.txt | awk '{print $1}' | sort | uniq  | awk '
-  BEGIN {
-    print "<eps> 0";
-  }
-  {
-    if ($1 == "<s>") {
-      print "<s> is in the vocabulary!" | "cat 1>&2"
-      exit 1;
-    }
-    if ($1 == "</s>") {
-      print "</s> is in the vocabulary!" | "cat 1>&2"
-      exit 1;
-    }
-    printf("%s %d\n", $1, NR);
-  }
-  END {
-    printf("#0 %d\n", NR+1);
-    printf("<s> %d\n", NR+2);
-    printf("</s> %d\n", NR+3);
-  }' > $dir/words.txt || exit 1;
+# NEW: for word classes
+word_class_command_line_option=""
+if [[ ! -z $word_class_dir ]]; then
+  word_class_command_line_option="-remove_class_prefix "C=" -word_class_list $word_class_dir/word_class_list.txt"
+fi
+
+# The following call to the perl script utils/create_words_symbol_mapping.pl
+# replaces the awk expression to create '$dir/words.txt'; the awk expression
+# started with "cat $tmpdir/lexiconp.txt | awk '{print $1}' | sort | uniq  | awk ' ..."
+#
+# The main (and hopefully only) difference is that the perl script also creates
+# entries for word-class-disambiguation symbols in words.txt if word classes
+# are used.
+#
+# The script also creates entries for the <s> and </s> symbols (beginning and
+# end of sentences) as the awk expression did.
+utils/create_words_symbol_mapping.pl -lexp_file $tmpdir/lexiconp.txt $word_class_command_line_option \
```

Comment:
The handling of command line options for all perl scripts has been changed to conform with the Kaldi (script) coding conventions. I tried to use 'egs/wsj/s5/utils/write_kwslist.pl' as a template where possible.